// menubar/lib/util/cleanOptions.spec.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var path = __importStar(require("path"));
var electron_1 = require("../__mocks__/electron");
var cleanOptions_1 = require("./cleanOptions");
var DEFAULT_OPTIONS = {
    browserWindow: {
        height: 400,
        width: 400
    },
    dir: path.resolve(electron_1.MOCK_APP_GETAPPPATH),
    index: "file://" + path.join(path.resolve(electron_1.MOCK_APP_GETAPPPATH), 'index.html'),
    tooltip: ''
};
describe('cleanOptions', function () {
    it('should handle undefined', function () {
        expect(cleanOptions_1.cleanOptions(undefined)).toEqual(DEFAULT_OPTIONS);
    });
    it('should handle a string with relative path', function () {
        expect(cleanOptions_1.cleanOptions({ dir: 'MY_RELATIVE_PATH' })).toEqual(__assign(__assign({}, DEFAULT_OPTIONS), { dir: path.resolve('MY_RELATIVE_PATH'), index: "file://" + path.join(path.resolve('MY_RELATIVE_PATH'), 'index.html') }));
    });
    it('should handle a string with absolute path', function () {
        expect(cleanOptions_1.cleanOptions({ dir: '/home/me/MY_ABSOLUTE_PATH' })).toEqual(__assign(__assign({}, DEFAULT_OPTIONS), { dir: '/home/me/MY_ABSOLUTE_PATH', index: 'file:///home/me/MY_ABSOLUTE_PATH/index.html' }));
    });
    it('should handle an object', function () {
        expect(cleanOptions_1.cleanOptions({
            browserWindow: {
                height: 100
            },
            index: 'file:///home/abc/index.html',
            showDockIcon: true,
            windowPosition: 'trayCenter'
        })).toEqual(__assign(__assign({}, DEFAULT_OPTIONS), { browserWindow: __assign(__assign({}, DEFAULT_OPTIONS.browserWindow), { height: 100 }), index: 'file:///home/abc/index.html', showDockIcon: true, windowPosition: 'trayCenter' }));
    });
});
